### ISSUE TITLE
Module Proposal: SVG Uploads

### ISSUE THREAD

**Issue by masteradhoc** [WordPress/performance#410]
## Overview
- Proposed module name: SVG Uploads
- Proposed module slug: svg-uploads
- Proposed module owner GitHub username(s):

### Purpose

Purpose of this module is to allow users to upload SVG (Scalable Vector Graphics) that dont contain script tags. This should allow people to use the big benefit of SVG Images:  unlimited scaling by staying always sharp with a very low file size.

When (trying to) upload a SVG to the media library that includes scripts we should give the user a notice that uploading SVGs containing scripts are currently not supported in core. (See scope) A Error Notice could also include a information for the user about potential security issues with SVGs that include scripts.

### Scope
In the first phase the suggestion is to allow uploading of regular SVGs without scripts. Additionally a SVG Preview should be available in the media library. 

Ability to upload SVGs that contain scripts is out of scrope.

### Rationale

SVGs are used for all types of vectorized images you need on your website. Vectorized Images are not created by the majority of people themself. Mostly they are prepared and generated by graphical designers and serve for a lot different options. Mostly they are used for logos, favicons, icons and illustrations. When uploading currently users are frustrated as core doesnt allow svgs by now. Users have to get back to the graphical designer to send the image again in a (maybe in that case less appropriate) filetype like jpg or png.

SVGs are widely supported by all browsers and is pretty much the standard for vectorized images: ["svg" | Can I use... Support tables for HTML5, CSS3, etc](https://caniuse.com/?search=svg)

### Other

SVGs are XML based and a vectorized image format. A deeper explaination of SVG and the usage can be found in this blog article: [What is an SVG File Used For and Why Developers Should be Using Them | Delicious Brains](https://deliciousbrains.com/svg-advantages-developers/)

A really long time ago a ticket in trac was opened by JustinSainton [#24251 Reconsider SVG inclusion to get_allowed_mime_types – WordPress Trac](https://core.trac.wordpress.org/ticket/24251). The ticket gained some popularity initially and even over time people commented about the ongoing need for a native svg integration. A big discussion and debate inside the ticket and a lot chats happened due to the security risks with scripted SVGs.

In the issue #357 @Zodiac1978 after a discussion with @spacedmonkey pointed out that a possible first step could be to start with static SVGs first. This approach could now be discussed in this Module Proposal.

As pointed out SVG Integration are already quite a topic for site owners seeing over 1 Million active Installations currently: [Suchergebnisse für „svg upload“ | WordPress.org Deutsch](https://de.wordpress.org/plugins/search/svg+upload/). Every week these plugins gain way over 150k new downloads.

## Current Implementations available
Tried to search around to see what is already available as a starting point. Note: some offer way more features than we need for step 1.

- (2017) Patch from Luke Cavanagh from the original trac ticket: https://core.trac.wordpress.org/attachment/ticket/24251/24251.2.diff
- different plugins https://de.wordpress.org/plugins/search/svg+upload allow currently sanitazion of svgs. The big three plugins available [Save SVG](https://de.wordpress.org/plugins/safe-svg/), [SVG Support](https://wordpress.org/plugins/svg-support/), [WP SVG Images](https://de.wordpress.org/plugins/wp-svg-images) all currently use the SVG Sanitizer you'll see below
- SVG Sanitize Library by @darylldoyle: https://github.com/darylldoyle/svg-sanitizer is used quite a lot as we saw in various wordpress plugins above but also in other CMS Implementations like [Drupal](https://www.drupal.org/project/svg_sanitizer) and [TYPO3](https://github.com/TYPO3GmbH/svg_sanitizer) no

**Comment by mxbclang** [comment]
Thank you, @masteradhoc! If you wouldn't mind pinging me in this issue when the proposal is all ready to go, we can work together to decide when to discuss this in an upcoming Slack chat.

**Comment by mxbclang** [comment]
Original discussion: #357

**Comment by mxbclang** [comment]
Noting that this proposal was discussed in our <a href="https://make.wordpress.org/core/2022/07/12/performance-team-meeting-summary-11-july-2022/">11 July performance chat</a> and will be moving forward! 🥳 

@masteradhoc will work to get started on this but will need engineering help. Brian, please feel free to ping me in this issue or in Slack when you need assistance and I can coordinate with the broader team.

**Comment by mxbclang** [comment]
@spacedmonkey Wanted to check in with you as you expressed some interest in this one in our chat last week. Would you be interested in being a "co-owner" of this module and taking the lead on the engineering? Your expertise and experience with this on Web Stories is definitely helpful here! cc @felixarntz

**Comment by spacedmonkey** [comment]
> Wanted to check in with you as you expressed some interest in this one in our chat last week. Would you be interested in being a "co-owner" of this module and taking the lead on the engineering? Your expertise and experience with this on Web Stories is definitely helpful here! cc @felixarntz

@bethanylang No, but for couple of reasons. 1. I am taking the next 6 weeks off and will not be on this project. 2. I am currently leading other efforts, like dominate color. 

I willing to support with code review and testing. 

If I find some time, I may look into prototyping a module. But leading the efforts to get this into core, it more than I have time for at the moment.

**Comment by mxbclang** [comment]
Thanks @spacedmonkey and thanks for sharing this in the #core-media channel!

**Comment by jeffpaul** [comment]
Worth noting here that Safe SVG is backed by the https://github.com/darylldoyle/svg-sanitizer library which you may find useful in your research/recommendations here.

**Comment by masteradhoc** [comment]
Thanks @spacedmonkey & @jeffpaul. Maybe i have a good announcement soon with a joining member :) . Will keep you posted. In the meantime if someone is interested to join let me know.

**Comment by mxbclang** [comment]
@masteradhoc To keep this in line with our other modules, I'm going to close out this issue as the initial proposal and I've created a new "epic" issue for this module here: #427. In the past, we've used the "epic" issue as a kind of parent issue to list out any other issues that are required to complete the module; for example, see #22. Totally up to you and any other developers who work on this if they'd like to use this in the same way.  :)

Closing this – see everyone in #427!
